src/webui/api/apicontroller.cpp

```diff
@@ -37,8 +37,9 @@
 
 #include "apierror.h"
 
-APIController::APIController(QObject *parent)
-    : QObject {parent}
+APIController::APIController(IApplication *app, QObject *parent)
+    : QObject(parent)
+    , ApplicationComponent(app)
```

Comment:
It seems unused, is it necessary to add it?
If possible I would like `ApplicationComponent` to be added to only where it is actually used.

Reply:
> It seems unused, is it necessary to add it?

It is used by subclasses.
And I really don't see the point why one `APIController` is an `ApplicationComponent` and the other is not (even if it doesn't need to use its capabilities yet).
Unnatural restrictions do no good.